src/boot/lib/pprint.ml

```diff
@@ -143,7 +184,7 @@ let rec ustring_of_ty = function
   | TyVariant _ ->
       failwith "Printing of non-empty variant types not yet supported"
   | TyVar (_, x, s) ->
-      ustring_of_var x s
+      ustring_of_con x s
```

Comment:
How come you use `ustring_of_con` here? In `pprint.mc` the identifier of a `TyVar` is printed using `pprintEnvGetStr` which does not, as far as I can tell, prepend invalid names with `#con`. You also use `ustring_of_con` for `PatCon`, `TmConDef` and `TmConApp`, but their constructor identifiers are printed using `pprintConName` which does prepend `#con`.

This question also applies to the pretty-printing of `TmType`.

Reply:
Thanks! I thought `TyVar` and `TmType` were escaped as well, but I should have checked.